tests/unit/features/DenseSubSamplesFeatures_unittest.cc

```diff
@@ -440,3 +440,91 @@ TEST(DenseSubSamplesFeatures, test6)
 	SG_UNREF(features_train2);
 	SG_UNREF(latent_features_train2);
 }
+
+TEST(DenseSubSamplesFeatures, test7)
+{
+	index_t n=6;
+	index_t dim=2;
+	index_t m=4;
+	float64_t rel_tolorance=1e-10;
+	float64_t abs_tolorance;
+
+	SGMatrix<float64_t> feat_train(dim, n);
+	SGMatrix<float64_t> lat_feat_train(dim, m);
+
+	feat_train(0,0)=0.81263;
+	feat_train(0,1)=0.99976;
+	feat_train(0,2)=1.17037;
+	feat_train(0,3)=1.51752;
+	feat_train(0,4)=1.57765;
+	feat_train(0,5)=3.89440;
+
+	feat_train(1,0)=0.5;
+	feat_train(1,1)=0.4576;
+	feat_train(1,2)=5.17637;
+	feat_train(1,3)=2.56752;
+	feat_train(1,4)=4.57765;
+	feat_train(1,5)=2.89440;
+
+	lat_feat_train(0,0)=1.00000;
+	lat_feat_train(0,1)=3.00000;
+	lat_feat_train(0,2)=4.00000;
+	lat_feat_train(0,3)=-8.00000;
+
+	lat_feat_train(1,0)=3.00000;
+	lat_feat_train(1,1)=2.00000;
+	lat_feat_train(1,2)=5.00000;
+	lat_feat_train(1,3)=-7.00000;
+
+	SGMatrix<float64_t> feat_train2(dim, n/2);
+	feat_train2(0,0)=0.81263;
+	feat_train2(0,1)=1.17037;
+	feat_train2(0,2)=1.57765;
+
+	feat_train2(1,0)=0.5;
+	feat_train2(1,1)=5.17637;
+	feat_train2(1,2)=4.57765;
+
+	CDenseFeatures<float64_t>* features_train0=new CDenseFeatures<float64_t>(feat_train2);
+	CDenseFeatures<float64_t>* latent_features_train0=new CDenseFeatures<float64_t>(lat_feat_train);
+
+	SGVector<int32_t> idx1(n/2);
+	for(int i=0; i<n; i++)
+	{
+		if (i%2==0)
+			idx1[i/2]=i;
+	}
+	CDenseFeatures<float64_t>* features_train1=new CDenseFeatures<float64_t>(feat_train);
+	SG_REF(features_train1);
+	CDenseSubSamplesFeatures<float64_t>* features_train2=new CDenseSubSamplesFeatures<float64_t>(features_train1, idx1);
+
+
+	float64_t ell=0.5;
+	CKernel* kernel=new CGaussianKernel(10, 2*ell*ell);
+
+	SG_REF(features_train0);
+	SG_REF(latent_features_train0);
+	kernel->init(features_train0, latent_features_train0);
+	SGMatrix<float64_t> res0=kernel->get_kernel_matrix();
+
+	SG_REF(features_train2);
+	SG_REF(latent_features_train0);
+	kernel->init(features_train2, latent_features_train0);
+	SGMatrix<float64_t> res1=kernel->get_kernel_matrix();
+
+	for(index_t i=0; i<res1.num_rows; i++)
```

Comment:
What about having a Macro to compare matrices with EXPECT/ASSERT_EQUAL?

Where would we add this?
@lisitsyn do you have an opinion on this?